xmtp_mls/src/groups/mod.rs

```diff
@@ -203,9 +218,20 @@ where
         mls_group.save(provider.key_store())?;
 
         let group_id = mls_group.group_id().to_vec();
-        let stored_group = StoredGroup::new(group_id.clone(), now_ns(), membership_state);
+        let stored_group = StoredGroup::new(
+            group_id.clone(),
+            now_ns(),
+            membership_state,
+            added_by_address.clone(),
+        );
+
         stored_group.store(provider.conn())?;
-        Ok(Self::new(client, group_id, stored_group.created_at_ns))
+        Ok(Self::new(
+            client,
+            group_id,
+            stored_group.created_at_ns,
+            added_by_address,
```

Comment:
Ethan and I paired and got on the same page. My initial hesitation was that I didn't understand why we needed to set the address on the `MlsGroup::new` and also on the `StoredGroup:new` seems like we should just be able to set it on the stored group and then fetch it from the stored group? Came to the conclusion that we don't do anything with the storedGroup in the bindings so this seems like the best way given our current architecture cc @richardhuaaa if I'm missing some context.

Reply:
@nplasterer it's a good point. Today, we're fetching the StoredGroup, and then manually caching the values onto the MlsGroup via the constructor so that we don't need to re-fetch it again. The drawback is that the constructor for MlsGroup is getting more complicated over time. Could be refactored one of two ways:

1. Don't cache the values onto the MlsGroup. This is simpler code, but tradeoff is you need to re-fetch from the DB every time you need a property like `added_by`.

2. Instead of manually copying the values over to the MlsGroup, have the MlsGroup hold a copy of StoredGroup inside it so there's no duplication. That's probably the best of both worlds.

Reply:
I like 2. do you think this will be hard to do later/a breaking change? Just seeing how we should prioritize that work. My gut is saying we should probably do it now before we filter this through the sdks.